stable/fluentd/templates/deployment.yaml

```diff
@@ -14,12 +14,14 @@ metadata:
 spec:
 {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.replicaCount }}
+{{- else}}
+  serviceName: {{ template "fluentd.name" . }}
```

Comment:
can you explain why we add this else for the `serviceName` ?

Reply:
It's [required by the spec](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.13/#statefulsetspec-v1-apps).

Reply:
I came from the other direction, deploying the chart threw an error indicating that `serviceName` was missing from the StatefulSet manifest.

About using `else` specifically,  convenience since the `if` already tests the right condition. Perhaps it is clearer to use an explicit `if` here, what do you think?

Reply:
@gellardo it improves the readability of this code